waspc/cli/src/Wasp/Cli/Command/CreateNewProject.hs

```diff
@@ -18,27 +18,42 @@ import Wasp.Cli.Command (Command, CommandError (..))
 import qualified Wasp.Cli.Command.Common as Command.Common
 import qualified Wasp.Cli.Common as Common
 import qualified Wasp.Data
-import Wasp.Util (indent)
+import Wasp.Util (indent, kebabToCamelCase)
 import qualified Wasp.Util.Terminal as Term
 
-newtype ProjectName = ProjectName {_projectName :: String}
+data ProjectInfo = ProjectInfo
+  { _projectName :: String,
+    _appName :: String
+  }
+
+-- Takes a project name String
+-- Returns either the ProjectInfo type that contains both the Project name
+-- and the App name (which might be the same), or an error describing why the name is invalid
+parseProjectNameIntoProjectInfo :: String -> Either String ProjectInfo
+parseProjectNameIntoProjectInfo name
+  | isValidWaspIdentifier appIdentifier = Right (ProjectInfo name appIdentifier)
+  | otherwise =
+      Left $
+        intercalate
+          "\n"
+          [ "The project's name is not of a valid format!",
+            indent 2 "- It can start with a letter or an underscore.",
+            indent 2 "- It can contain only letters, numbers, dash, or underscores.",
+            indent 2 "- It can't be a Wasp keyword."
+          ]
+  where
+    appIdentifier = kebabToCamelCase name
```

Comment:
As described in [an earlier comment](https://github.com/wasp-lang/wasp/pull/700#discussion_r955900024), we should change this variable to `appName`

Reply:
fixed